Core/SqlStatementModel/ConstantCollectionExpression.cs

```diff
@@ -0,0 +1,119 @@
+﻿// This file is part of the re-linq project (relinq.codeplex.com)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// re-linq is free software; you can redistribute it and/or modify it under
+// the terms of the GNU Lesser General Public License as published by the
+// Free Software Foundation; either version 2.1 of the License,
+// or (at your option) any later version.
+//
+// re-linq is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-linq; if not, see http://www.gnu.org/licenses.
+//
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using Remotion.Utilities;
+
+namespace Remotion.Linq.SqlBackend.SqlStatementModel
+{
+  /// <summary>
+  /// Expression representing an <see cref="ICollection"/>, an <see cref="ICollection{T}"/> or an <see cref="IReadOnlyCollection{T}"/>.
+  /// </summary>
+  public class ConstantCollectionExpression : Expression
+  {
+    /// <summary>
+    /// The collection that was the original <see cref="ConstantExpression"/>'s <see cref="ConstantExpression.Value"/>.
+    /// </summary>
+    public object Collection { get; }
```

Comment:
design-question: wouldn't it be better to use 
```suggestion
    public object Collection => _items;
```
(can be done as expression or statement.

We save one pointer.
Also, it getse moved down to GetItems with this change.